lms/djangoapps/teams/views.py

```diff
@@ -485,8 +487,20 @@ def has_object_permission(self, request, view, obj):
             """Returns true if the user is enrolled or is staff."""
             return has_team_api_access(request.user, obj.course_id)
 
+    class IsStaffOrTAOrReadOnly(IsStaffOrReadOnly):
+        """Permission that checks to see if the user is global staff, course
+        staff, or has discussion privileges. If none of those conditions are
+        met, only read access will be granted.
+        """
+
+        def has_object_permission(self, request, view, obj):
+            return (
+                has_discussion_privileges(request.user, obj.course_id) or
+                IsStaffOrReadOnly.has_object_permission(self, request, view, obj)
```

Comment:
I would use super in all cases of trying to invoke the parent class behavior.  "MI isn't in use here" is logic that assumes you know every way this class will ever be subclassed.  By using super(), you write a class that will work no matter what happens later.

Also, I wonder about the need for nested classes.  There's no advantage to nesting them, and it makes the class harder to get at (as we have seen).  The usual argument is, "Only this outer class needs to use the inner class, so it should be hidden inside it."  This is a "Java mindset" way of thinking, and can hinder things like introspection, testing, and tooling.  Just making it a peer of the class that needs it is usually simpler.